app/controllers/carts_controller.rb

```diff
@@ -49,6 +49,7 @@ def apply_coupon
     if coupon_exists?(params[:coupon])
       @coupon = get_discount(params[:coupon])
       session[:coupon] = @coupon
+      session[:coupon_id] = Coupon.find_by(name: params[:coupon]).id
```

Comment:
I think there should be something like `Coupon.find_by(name: params[:coupon])&.id`